aten/src/ATen/native/MaxUnpooling.cpp

```diff
@@ -14,16 +14,16 @@ Tensor& max_unpooling2d_forward_out_cpu(
   auto owidth = output_size[1];
   TORCH_CHECK(
       indices_.scalar_type() == at::ScalarType::Long,
-      "elements in indices should be type int64");
+      "elements in indices should be type int64 but got: ", indices_.scalar_type());
   TORCH_CHECK(
       output_size.size() == 2,
-      "There should be exactly two elements (height, width) in output_size");
+      "There should be exactly two elements (height, width) in output_size, but got ", output_size.size(), " elements.");
   TORCH_CHECK(
       (self_.ndimension() == 3 || self_.ndimension() == 4),
-      "Input to max_unpooling2d should be a 3d or 4d Tensor");
+      "Input to max_unpooling2d should be a 3d or 4d Tensor, but got tensor with dimension: ", self_.ndimension());
   TORCH_CHECK(
       self_.sizes() == indices_.sizes(),
-      "Shape of indices should match shape of input");
+      "Expected shape of indices to be: ", self_.sizes(), " but got: ", indices_.sizes());
```

Comment:
nit: we lost the information that the expected size is the input's size no?

Reply:
Ah, right, good catch Alban. Thanks.

The error message now looks like below:

```python
RuntimeError: Expected shape of indices to be same as that of the input tensor ([1, 1, 2, 2]) but got indices tensor with shape: [2, 2, 1, 1]
```

Do you think this is better?

Reply:
Yep thanks!